docs/migration-guide-v2.md

```diff
@@ -0,0 +1,152 @@
+# Migration guide: v2.0.0
+
+Version 2.0.0 introduces some breaking changes. This page aims to help you migrate to this new major version. The relevant changes are:
+
+- The deprecated `connect` and `disconnect` methods have been removed
+- The deprecated `filtered_messages` and `unfiltered_messages` methods have been removed
+- User-managed queues for incoming messages have been replaced with a single client-wide queue
+- Some arguments to the `Client` have been renamed or removed
+
+## Changes to the client lifecycle
+
+The deprecated `connect` and `disconnect` methods have been removed. The best way to connect and disconnect from the broker is through the client's context manager:
+
+```python
+import asyncio
+import aiomqtt
+
+
+async def main():
+    async with aiomqtt.Client("test.mosquitto.org") as client:
+        await client.publish("temperature/outside", payload=28.4)
+
+
+asyncio.run(main())
+```
+
+If your use case does not allow you to use a context manager, you can use the client’s `__aenter__` and `__aexit__` methods almost interchangeably in place of the removed `connect` and `disconnect` methods.
+
+The `__aenter__` and `__aexit__` methods are designed to be called by the `async with` statement when the execution enters and exits the context manager. However, we can also execute them manually:
+
+```python
+import asyncio
+import aiomqtt
+
+
+async def main():
+    client = aiomqtt.Client("test.mosquitto.org")
+    await client.__aenter__()
+    await client.publish("temperature/outside", payload=28.4)
+    await client.__aexit__(None, None, None)
```

Comment:
Maybe use a try except block as suggested in `connecting-to-the-broker.md`.

Reply:
Good idea, I put `__aenter__` outside the `try` clause, as in [PEP 343](https://peps.python.org/pep-0343/), I hope that gets it right 😅